rclcpp/src/rclcpp/expand_topic_or_service_name.cpp

```diff
@@ -133,9 +133,13 @@ rclcpp::expand_topic_or_service_name(
           RCL_RET_ERROR, "failed to validate node name",
           rmw_get_error_state(), rmw_reset_error);
       }
+
+      const char * validation_result_string = rmw_node_name_validation_result_string(
+        validation_result);
+      validation_result_string = validation_result_string ? validation_result_string : "";
```

Comment:
While using the ternary operator isn't exactly wrong, it is definitely odd to be re-assigning a variable back to itself.  I'd prefer a simple `if (nullptr == validation_result_string)` here.  Also, we should probably have a slightly different string than `""`; maybe something like `"Unknown error"`.

The same goes for the rest of the calls below.

Reply:
actually, I've ever tried to make the error string specific, but when the result is valid, it also return NULL which is different from the default NULL in some sense, so it's not accurate if we treat all the NULL to "unknown error", I prefer to tweak the `rmw_*validate_result_string` to get it more accurate, any comments ? thanks.

Reply:
Yeah, I think I agree that we should ensure that the `rmw_*_validation_result_string` should probably never return NULL.  I'll make a similar comment over in ros2/rmw#130 , but let's leave this open for now just in case we come to a different conclusion over there.